src/overview/actions.js

```diff
@@ -126,3 +130,23 @@ export const deleteDocs = () => async (dispatch, getState) => {
     // Refresh search view after deleting all assoc docs
     dispatch(search({ overwrite: true }))
 }
+
+export const unBookmark = () => async (dispatch, getState) => {
+    const url = selectors.bookmarkInfoToSend(getState())
+    console.log(url)
+
+    await removeBookmarkByUrl(url)
+
+    // Refresh search view after unbookmarking
+    dispatch(search({ overwrite: true }))
+}
+
+export const bookmark = () => async (dispatch, getState) => {
+    const url = selectors.bookmarkInfoToSend(getState())
+    console.log(url)
+
+    await createBookmarkByExtension(url)
+
+    // Refresh search view after bookmarking
+    dispatch(search({ overwrite: true }))
+}
```

Comment:
You should be able to unify these two actions into one (like "toggleBookmark") and just use the current state to switch behaviour between calling `createBookmarkByExtension` or `removeBookmarkByUrl`.

Another thing you could do: pass the URL in as an arg instead of storing it in redux state. Then you don't need the `setBookmarkUrl` action and all the state associated with it.

Reply:
When I say "just use the current state to switch behaviour", this would need to pass in the index as an arg, then you just use `getState` and a selector to grab the current state at that index.